src/main/scala/thurloe/crypto/Encryption.scala

```diff
@@ -0,0 +1,79 @@
+package thurloe.crypto
+
+import java.security.SecureRandom
+import javax.crypto.spec.{IvParameterSpec, SecretKeySpec}
+import javax.crypto.Cipher
+
+import org.apache.commons.codec.binary.Base64
+
+import scala.util.{Failure, Success, Try}
+
+sealed trait Encryption {
+
+  def encryption: String
+  def blockSize: Int
+  def keySize: Int
+  def paddingMode: String
+  def cipherMode: String
+
+  val ranGen = new SecureRandom()
+
+  final def init(mode: Int, secretKey: Array[Byte], iv: Array[Byte]) = {
+    val key = new SecretKeySpec(secretKey, encryption)
+    val ivSpec = new IvParameterSpec(iv)
+    val cipher = Cipher.getInstance(s"$encryption/$cipherMode/$paddingMode")
+    cipher.init(mode, key, ivSpec)
+    cipher
+  }
+
+  final def encrypt(plainText: Array[Byte], secretKey: SecretKey): Try[EncryptedBytes] = {
+    if(secretKey.key.length != keySize / 8) {
+      Failure(new IllegalArgumentException(s"Key size (${secretKey.key.length}) did not match required ${keySize / 8}"))
+    }
+    else {
+      // Generate an IV:
+      val iv = new Array[Byte](blockSize / 8)
+      ranGen.nextBytes(iv)
+
+      val cipher = init(Cipher.ENCRYPT_MODE, secretKey.key, iv)
+      Success(EncryptedBytes(cipher.doFinal(plainText), iv))
+    }
+  }
+
+  final def decrypt(encryptedBytes: EncryptedBytes, secretKey: SecretKey): Try[Array[Byte]] = {
+    if(secretKey.key.length != keySize / 8) {
```

Comment:
This is the same check as above. Can you pull that into a function or something like that?